src/DataProtection/DataProtection/src/DataProtectionBuilderExtensions.cs

```diff
@@ -223,6 +224,7 @@ public static IDataProtectionBuilder PersistKeysToFileSystem(this IDataProtectio
         /// <param name="builder">The <see cref="IDataProtectionBuilder"/>.</param>
         /// <param name="registryKey">The location in the registry where keys should be stored.</param>
         /// <returns>A reference to the <see cref="IDataProtectionBuilder" /> after this operation has completed.</returns>
+        [SupportedOSPlatform("windows")]
```

Comment:
👍

Reply:
Out of interest, is it necessary to apply it to all of the members, or could it just be applied to the type itself?

Reply:
cc @blowdart to also review that this is all Windows only surface area for data protection

Reply:
The registry one? Yes. What is this attribute actually doing?

Reply:
> What is this attribute actually doing?

Registering for the Platform Compatibility Analyzer that this API will only work on windows, and callers that are platform-neutral or supporting other platforms would see an analyzer warning when referencing this API.

Reply:
@blowdart are the UseCustomCryptographicAlgorithms methods also windows only?  Or are we just doing the minimal set that gets rid of analyzer warnings at this point?

https://github.com/dotnet/aspnetcore/pull/25421/files#diff-2ee1ae56bd7bb17bbf8807958183a2adR520

Reply:
The SDK adds an analyzer that tells you when you use APIs that are compatible with certain platforms. The runtime annotated their API surfaces and we're propagating it further up. See https://github.com/dotnet/designs/blob/master/accepted/2020/platform-exclusion/platform-exclusion.md for more about this.

Reply:
OK, for the registry and DPAPI bits that's fine. 

The CngCustomCrypto is windows only, the others are not. So you should be able to flag

 public static IDataProtectionBuilder UseCustomCryptographicAlgorithms(this IDataProtectionBuilder builder, CngCbcAuthenticatedEncryptorConfiguration configuration)

public static IDataProtectionBuilder UseCustomCryptographicAlgorithms(this IDataProtectionBuilder builder, CngGcmAuthenticatedEncryptorConfiguration configuration)

And those configuration classes too.

Reply:
👍 Filed https://github.com/dotnet/aspnetcore/issues/25781 to do a follow up. Just trying to avoid respinning a build on this PR if I can